content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -0,0 +1,920 @@
+
+## WGPU Upgrade and Renderer Improvements
+
+<div class="release-feature-authors">authors: @Elabajaba</div>
+
+In Bevy 0.13 we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to do things like compiling shaders asynchronously to avoid shader compilation stutters, and to multithread draw call creation for better performance.
+
+Due to changes in wgpu 0.19, we've added a new `webgpu` feature to Bevy that is now required when doing WebAssembly builds targeting WebGPU. Disabling the `webgl2` feature is no longer required when targeting WebGPU, but the new `webgpu` feature currently overrides the `webgl2` feature when enabled. Library authors, please do not enable the `webgpu` feature by default. In the future we plan on allowing you to target both WebGL2 and WebGPU in the same WebAssembly binary, but it requires reworking parts of the renderer where we're relying on compile time constants when targeting `webgl2`, and adding a way to choose the renderer's backend at runtime on web.
+
+As usual, there's been some changes that may cause issues for custom shaders. We've swapped the material and mesh bind groups, so that mesh data is now in bind group 1, and material data is in bind group 2. This greatly improved our draw call batching when combined with changing the sorting functions for the opaque passes to sort by pipeline and mesh. Previously we were sorting them by distance from the camera. These batching improvements mean we're doing fewer draw calls, which improves CPU performance, especially in larger scenes. We've also removed the `get_instance_index` function in shaders, as it was only required to workaround an upstream bug that has been fixed in wgpu 0.19. For other shader or rendering changes, please see the [migration guide](/learn/migration-guides/0.12-0.13/) and [wgpu's changelog](https://github.com/gfx-rs/wgpu/blob/v0.19/CHANGELOG.md).
+
+## Texture Atlas Rework
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Sprite Slicing and Tiling
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Exposure Settings
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Minimal Reflection Probes
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Light Maps
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Light `RenderLayers`
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Approximate Indirect Specular Occlusion
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Unload Render Assets From RAM
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Bind Group Layout Entries
+
+<div class="release-feature-authors">authors: @TODO</div>
+
+TODO.
+
+## Type-Safe Labels for the `RenderGraph`
```

Comment:
There might be a mistake made while merging. This title should be `## Camera-Driven UI`; and where should be this title, I don't know.